sherpa/astro/io/crates_backend.py

```diff
@@ -307,99 +318,121 @@ def _try_key_list(crate, keyname, num, dtype=SherpaFloat, fix_type=False):
 
     # Make a copy of the data, since we don't know that pycrates will
     # do something sensible wrt reference counting
-    key = numpy.array([key.value] * num)
+    key = np.array([key.value] * num)
```

Comment:
There's always the balance between speed and convenience (e.g. I don't **myself** think that the I/O routines are time critical, but I tend not to be using large files). 

Saying that, I'm not sure that `[foo] * num` is more or less reasable than using `np.full`!

Reply:
I actually benchmarked that. For this one line, the different in runtime is about a factor of 5-10 in speed for a column with 1024 entries and bigger for bigger files. It's still >> 1s so it's certainly not critical because I/O routines tend to be limited by disk I/O anyway. I would not open a new PR just for this. However, "sherpa performance" was on my SDS task list for 2023 and is again for 2024, so I figured I point it out when the code is touched anyway and it's just an easy change to do that is about equally readable.

Reply:
I still want the `np.fill...` here...

Reply:
and below where we have the same pattern.